docs/TuxMutableClass.md

```diff
@@ -0,0 +1,142 @@
+# TuxMutableClass
+
+## Table of Contents
+<ol>
+  <li><a href="#Premise">Premise</a></li>
+  <li><a href="#Implementation">Implementation</a>
+    <ol style="list-style-type: upper-alpha;">
+      <li><a href="#Requiring-TuxMutableClass">Requiring TuxMutableClass</a></li>
+      <li><a href="#creating-mutable-class-components">Creating Mutable Class Components</a></li>
+      <ol style="list-style-type: upper-roman;">
+        <li><a href="#create-mutable-class-example">Create Mutable Class Example</a></li>
+        <li><a href="#React-createMutableClass">METHOD: React.createMutableClass</a></li>
+        <li><a href="#mutableClass">OBJECT: mutableClass</a>
+          <ol style="list-style-type: lower-alpha;">
+            <li><a href="#mutableClass-componentWillMount">METHOD: mutableClass.componentWillMount</a></li>
+            <li><a href="#mutableClass-shouldComponentUpdate">METHOD: mutableClass.shouldComponentUpdate</a></li>
+          </ol>
+        </li>
+      </ol>
+    </ol>
+  </li>
+  <li><a href="#TuxMutableClass-Guidelines">TuxMutableClass Guidelines</a></li>
+  <li><a href="#TuxMutableClass-Complete-Example">TuxMutableClass Complete Example</a></li>
+</ol>
+
+## <a id="Premise"></a>Premise [#](#Premise)
+>The Tux Mutable Class is an opinionated type of React class designed compare specified mutable traits to determine if a component should re-render.
+
+## <a id="Implementation"></a>Implementation [#](#Implementation)
+Tux determines if a mutable class component should re-render by taking advantage of the `componentWillMount` and `shouldComponentUpdate` life cycle events. Specifically in the `componentWillMount` life cycle, the component will do a deep search of its `state` and `props` to create paths to the specified `mutableTraits`. The paths are then attached to the constructor's prototype, such that a deep search to create the paths only needs to be done once for a particular type of component.
+
+These `mutableTraits` paths are then used in the `shouldComponentUpdate` life cycle event to see if those properties have changed when comparing the current state and props to the next props and state of the component. The `shouldComponentUpdate` life cycle event determines if a component should re-render in the virtual DOM. By performing this check you can prevent the unneeded re-rendering of components within the virtual DOM each time potential re-render is triggered.
```

Comment:
Change second sentence to: If those state and/or props have not changed then `shouldComponentUpdate` will return false which indicates to React that it does not need to update the component in the DOM.

Add an `a` between `each time` and `potential`.